### ISSUE TITLE
[FEATURE REQUEST] Local data encryption

### ISSUE THREAD

**Issue by domportera** [Akylas/OSS-DocumentScanner#262]
**Is your feature request related to a problem? Please describe.**
Quite often these applications are used to scan highly sensitive documents (esp Card Wallet) - existing on-device unencrypted can be a security risk

**Describe the solution you'd like**
[Local database encryption](https://source.android.com/docs/security/features/encryption/file-based) by default for scanned-but-not-exported documents/cards
option to export PDFs with password
the easiest development-wise might be to have the user be able to set a global password, stored locally (encrypted via fingerprint/android unlock) and have this global password apply to everything exported

**Describe alternatives you've considered**
An application to encrypt once exported, though no alternative solution can exist for the apps' internal scans outside

**Additional context**
I'm a big fan of OSS card wallet and my suggestion largely stems from that, as I am keeping these files for as long as my phone lives. I'd like it to be as secure as possible

I have not checked the source to see if you using file-based encryption already for application files, so ignore that part if you are!

Thanks!! Love your work

**Comment by farfromrefug** [comment]
@domportera thanks for the feature request. 
Right now the data is not encrypted and could be easily accessed. Files (images) are written on disk for optimized access and memory usage (no need to load full image in memory like i would have to do if the image was in the sqlite database).
Not sure how to solve this right now but it is a great feature idea.
I think if you choose in the settings to store data on the internal storage it would already be more secure as you would need to be root to access the data.

PS: the level of "security" right now is about the same as the photos in your gallery.

**Comment by domportera** [comment]
yeah that makes sense. really I'd say the level of security is at least more secure than the gallery 😅 DCIM folder permissions etc etc. So this isn't urgent.

It doesnt have to be a database implementation, I just assumed that's how you organized the cards. if the android file encryption system I linked isn't quite straightforward to seamlessly implement I get that - trading off usability for this security is likely not worth it.

however exporting/importing with a PDF password should be more straightforward I think, and would cover a fair amount of security concerns by itself for most users

**Comment by farfromrefug** [comment]
@domportera Yes i Will look at pdf with a password. Will also sync about encryption

**Comment by fbetil** [comment]
Hi, i'm interested to with local data encryption on wallet because even if i activate fingerprint access, images are directly accessible on SD cards.
Thanks a lot

**Comment by farfromrefug** [comment]
@fbetil i really like the idea. However it will require quite a bit of work, and right now without more sponsoring i can't work on big features like this.
Will leave this open. Will see what i can do later on

**Comment by fallsdevil** [comment]
I came here precisely to suggest the password option for PDFs! I don't usually use it, but I ended up coming across identity documents that I can't delete for a while, and it would be interesting to do the scanning and security in the same app. Furthermore, as they are talking about encryption and at the moment oss cannot implement it, perhaps an alternative would be ["image tool box"](https://github.com/T8RIN/ImageToolbox) despite being an app for images, it has the ability to encrypt any type of file based on a password. I tested it a few times and using images, the image does not appear in the gallery, in fact it has an error or incompatible icon. So maybe it will help with what they need.

**Comment by farfromrefug** [comment]
@fallsdevil @fbetil @domportera PDF password will be implemented in next version

**Comment by domportera** [comment]
u are epic 💪
